### PR TITLE
Allow icon_keyboard to be null

### PR DESCRIPTION
**Pull request by celotajstg**
## About The Pull Request

Some computers have icon_keyboard set to no_keyboard, which leads to a useless render of an empty sprite. We can do better than that.

## Why It's Good For The Game

The difference will be imperceptible as it won't reduce rendering calls much, but will be a useful code improvement.

## Changelog
:cl: Celotajs
code: Computers now have a proper way of not having keyboards instead of an empty sprite
/:cl:

### CODE REVIEW

**Comment by celotajstg** [comment]
Map compilation has failed as a result of a unrelated issue - for whatever reason it could not fetch byond properly.

**Comment by interestingusernam3** [comment]
Epic compilation fail

**Comment by maxymax13** [comment]
pay the compilation tax

File: code/game/machinery/computer/_computer.dm [hunk 1]
```diff
@@ -30,10 +30,11 @@
...
 	. = ..()
-	if(machine_stat & NOPOWER)
-		. += "[icon_keyboard]_off"
-		return
-	. += icon_keyboard
+	if(icon_keyboard)
+		if(machine_stat & NOPOWER)
+			. += "[icon_keyboard]_off"
+			return
+		. += icon_keyboard
```
Comment (LemonInTheDark) [inline code/game/machinery/computer/_computer.dm:37]: why not do an early return here?

Comment (Mothblocks) [reply code/game/machinery/computer/_computer.dm:37]: I agree.

Comment (celotajstg) [reply code/game/machinery/computer/_computer.dm]: Sure

File: code/game/machinery/computer/_computer.dm [hunk 2]
```diff
@@ -30,10 +30,10 @@
 
 /obj/machinery/computer/update_overlays()
 	. = ..()
-	if(machine_stat & NOPOWER)
-		. += "[icon_keyboard]_off"
-		return
-	. += icon_keyboard
+	if(icon_keyboard)
+		if(machine_stat & NOPOWER)
+			return . += "[icon_keyboard]_off"
```
Comment (Mothblocks) [inline code/game/machinery/computer/_computer.dm:35]: No no no, just do `. + "[icon_keyboard]_off"`.

File: code/game/machinery/computer/_computer.dm [hunk 3]
```diff
@@ -30,10 +30,10 @@
...
+		. += icon_keyboard
```
Comment (Mothblocks) [inline code/game/machinery/computer/_computer.dm:36]: Here too.

File: code/game/machinery/computer/_computer.dm [hunk 4]
```diff
@@ -30,10 +30,10 @@
 
 /obj/machinery/computer/update_overlays()
 	. = ..()
-	if(machine_stat & NOPOWER)
-		. += "[icon_keyboard]_off"
-		return
-	. += icon_keyboard
+	if(icon_keyboard)
```
Comment (Mothblocks) [inline code/game/machinery/computer/_computer.dm:33]: And make this `if(!icon_keyboard) return`.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/game/machinery/computer/_computer.dm (hunk 3 above)
Comment (celotajstg) [reply code/game/machinery/computer/_computer.dm:36]: An early return is not intended in the icon_keyboard if

File: code/game/machinery/computer/_computer.dm (hunk 4 above)
Comment (celotajstg) [reply code/game/machinery/computer/_computer.dm:33]: Same as above, an early return is not intended in the icon_keyboard if

**Pull request merged**